packages/wallet/src/PersonalWallet/PersonalWallet.ts

```diff
@@ -518,6 +518,11 @@ export class PersonalWallet implements ObservableWallet {
       utxo: this.utxo
     });
 
+    this.keyAgent
+      .derivePublicKey(keyManagementUtil.DREP_KEY_DERIVATION_PATH)
+      .then((ed25519PublicKeyHex) => (this.drepPubKey = ed25519PublicKeyHex))
```

Comment:
If the derive call fails it's pretty bad because there's no retry mechanism in place.
We could retry when user calls `getPubDrepKey` so we can return the drep key or an error.
```diff
-  this.keyAgent.derivePublicKey
+ this.getPubDrepKey()
// ....
+ async getPubDRepKey(): Ed25519PublicKeyHex {
+   if (!this.drepPubKey) {
+     this.drepPubKey = await this.keyAgent.derivePublicKey(...)
+   }
+  return this.drepPubKey;
+ }
```

Reply:
added in [2aedd12](https://github.com/input-output-hk/cardano-js-sdk/pull/936/commits/2aedd125d87e57aa01d82c17366f19ea9e910929)

Reply:
Please review @mirceahasegan, then @mchappell please squash and rebase, prior to my review